lib/rubygems/security/signer.rb

```diff
@@ -29,6 +29,36 @@ class Gem::Security::Signer
 
   attr_reader :digest_name # :nodoc:
 
+  ##
+  # Attemps to re-sign an expired cert with a given private key
+  def self.re_sign_cert(cert_path, private_key_path)
+    unless File.exist?(cert_path)
```

Comment:
We probably will want to check that the user has permissions to access the file as well

Reply:
I think that issue would attend to itself by raising an EPERM. I suppose we could handle it explicitly, but it feels unnecessary to me for now.

Reply:
I agree with @djberg96